books/xdoc/base.lisp

```diff
@@ -72,11 +72,13 @@
       (and long (not (stringp long))
            ":long is not a string (or nil)")
       (and pkg (or (not (stringp pkg)) (equal pkg "") (not (pkg-witness pkg)))
-           ":pkg is not a string that is a known package (or nil)")))
+           ":pkg is not a string that is a known package (or nil)")
+      (and code-source (not (stringp code-source))
+           ":code-source is not a string (or nil)")))
```

Comment:
see my other comment -- I think we should accept `("path" :dir :dirname)` as well as just a string here